### PR TITLE
cplp - 463 - get app user role with respective app

### PR DESCRIPTION
**Pull request by Akshay-Chaudhari-BMW**
get app user role with respective app

### CODE REVIEW

File: src/administration/CatenaX.NetworkServices.Administration.Service/BusinessLogic/UserBusinessLogic.cs [hunk 1]
```diff
@@ -484,27 +484,35 @@ public async Task<bool> ExecuteOwnCompanyUserPasswordReset(Guid companyUserId, s
             throw new NotFoundException($"Cannot identify companyId or shared idp : companyUserId {companyUserId} is not associated with the same company as adminUserId {adminUserId}");
         }
 
-        public Task<Pagination.Response<CompanyAppUserDetails>> GetOwnCompanyAppUsersAsync(Guid appId, string iamUserId, int page, int size)
+        public async Task<Pagination.Response<CompanyAppUserDetails>> GetOwnCompanyAppUsersAsync(Guid appId, string iamUserId, int page, int size)
         {
-            var appUsers = _portalRepositories.GetInstance<ICompanyAssignedAppsRepository>().GetOwnCompanyAppUsersUntrackedAsync(appId, iamUserId);
+            var companyId = await _portalRepositories.GetInstance<IUserRepository>().GetCompanyIdForIamUserUntrackedAsync(iamUserId).ConfigureAwait(false);
```
Comment (ntruchsess) [inline src/administration/CatenaX.NetworkServices.Administration.Service/BusinessLogic/UserBusinessLogic.cs:489]: we should avoid to go to the database more often than necessary. To get the roles associated with a specific appId that are actually assigned to users of the same company as the calling user I suggest the following query:

```
    public IQueryable<CompanyUserAssignedRole> GetOwnCompanyAppUserAssignedRolesUntrackedAsync(Guid appId, string iamUserId) =>
        _context.CompanyUserAssignedRoles
            .AsNoTracking()
            .Where(assignedRole => assignedRole.UserRole!.IamClient!.Apps.Any(app => app.Id == appId)
                && assignedRole.CompanyUser!.Company!.CompanyUsers.Any(companyUser => companyUser.IamUser!.UserEntityId == iamUserId));

```

**Review by ntruchsess — CHANGES_REQUESTED** [review]

File: src/portalbackend/CatenaX.NetworkServices.PortalBackend.DBAccess/Repositories/CompanyAssignedAppsRepository.cs [hunk 2]
```diff
@@ -44,13 +44,12 @@ public CompanyAssignedAppsRepository(PortalDbContext portalDbContext)
     public CompanyAssignedApp CreateCompanyAssignedApp(Guid appId, Guid companyId, AppSubscriptionStatusId appSubscriptionStatusId ) =>
         _context.CompanyAssignedApps.Add(new CompanyAssignedApp(appId, companyId, appSubscriptionStatusId)).Entity;
 
-    public IQueryable<CompanyUser> GetOwnCompanyAppUsersUntrackedAsync(Guid appId, string iamUserId) =>
-        _context.CompanyAssignedApps
+    public IQueryable<CompanyUserAssignedRole> GetOwnCompanyAppUsersUntrackedAsync(Guid appId, string iamUserId) =>
```
Comment (ntruchsess) [inline src/portalbackend/CatenaX.NetworkServices.PortalBackend.DBAccess/Repositories/CompanyAssignedAppsRepository.cs:47]: To match my comment above I suggest:

```
    public IQueryable<CompanyUser> GetOwnCompanyAppUsersUntrackedAsync(Guid appId, string iamUserId) =>
       _context.CompanyUsers
            .AsNoTracking()
            .Where(companyUser => companyUser.UserRoles.Any(userRole => userRole.IamClient!.Apps.Any(app => app.Id == appId))
                && companyUser.Company!.CompanyUsers.Any(companyUser => companyUser.IamUser!.UserEntityId == iamUserId));

```

**Review by ntruchsess — CHANGES_REQUESTED** [review]

File: src/administration/CatenaX.NetworkServices.Administration.Service/BusinessLogic/UserBusinessLogic.cs [hunk 3]
```diff
@@ -484,27 +484,27 @@ public async Task<bool> ExecuteOwnCompanyUserPasswordReset(Guid companyUserId, s
...
                     appUsers.CountAsync(),
-                    appUsers.OrderBy(companyUser => companyUser.Id)
+                    appUsers.OrderBy(companyUserAssignedRoles => companyUserAssignedRoles!.CompanyUser!.Id)
                         .Skip(skip)
                         .Take(take)
-                        .Select(companyUser => new CompanyAppUserDetails(
-                            companyUser.Id,
-                            companyUser.CompanyUserStatusId,
-                            companyUser.UserRoles.Select(userRole => userRole.UserRoleText))
+                        .Select(companyUserAssignedRoles => new CompanyAppUserDetails(
```
[the remark this answers is not in the document]
Comment (Akshay-Chaudhari-BMW) [reply src/administration/CatenaX.NetworkServices.Administration.Service/BusinessLogic/UserBusinessLogic.cs]: resolved

File: src/portalbackend/CatenaX.NetworkServices.PortalBackend.DBAccess/Repositories/CompanyAssignedAppsRepository.cs (hunk 2 above)
Comment (Akshay-Chaudhari-BMW) [reply src/portalbackend/CatenaX.NetworkServices.PortalBackend.DBAccess/Repositories/CompanyAssignedAppsRepository.cs]: resolved

**Review by ntruchsess — APPROVED** [review]

**Pull request merged**